lib/saml_idp/xml_security.rb

```diff
@@ -180,8 +193,8 @@ def validate_doc_embedded_signature(
           '//ds:SignedInfo',
           { 'ds' => DSIG }
         )
-          { 'ds' => DSIG }
-        end
+                               { 'ds' => DSIG }
+                             end
```

Comment:
rubocop did something very funky here, maybe the whole thing might be easier by pulling the `REXML::XPath.first` into a local?
```ruby
signature_elem = REXML::XPath.first(
  ...,
)
sig_namespace_hash = { 'ds' => DSIG } if signature_elem
```

Reply:
yeah probably a good idea, i have fought with this rubocop suggestion over and over again and it finally slipped through

Reply:
actually looking at this chunk of code below, this is pretty weird, right? 

`REXML::XPath.first(@sig_element, "//ds:SignedInfo", {"ds"=>DSIG})` returns the expected `Signature` or `ds:Signature` node. 

i'm wondering what scenarios we envisioned where the `sig_namespace_hash` evaluates to `nil` (and if that's what we want to do)

```ruby
        sig_namespace_hash = if REXML::XPath.first(
          @sig_element,
          '//ds:SignedInfo',
          { 'ds' => DSIG }
        )
          { 'ds' => DSIG }
        end

        # verify signature
        signed_info_element = REXML::XPath.first(
          @sig_element,
          '//ds:SignedInfo',
          sig_namespace_hash
        )
```

https://github.com/18F/saml_idp/blob/main/lib/saml_idp/xml_security.rb#L178-L191

Reply:
anyways, i fixed the formatting thing here lol: https://github.com/18F/saml_idp/pull/108/commits/4d6da48ecaabbe8be0d06577d37b3c559902ad6f

(i wonder if we should just make a namespace hash of all the namespaces listed in the hash, and just pass that in as the namespace hash whenever we are using the `REXML::XPath` class.)

Reply:
This seems like a workaround to validating that the element is namespaced, but there is a better way to do this since we probably don't really care about the namespace at all.

To get the `SignedInfo` element, irrespective of namespace, we could do  `//*[local-name()]='SignedInfo'`. This would be an improvement for a different PR though.